### PR TITLE
#82 Constants for database fields refactor

### LINKED ISSUE #82: REF-82: Use constants instead of strings for db variables
**Refactor Type:**
- [Replace Magic Literal](https://refactoring.com/catalog/replaceMagicLiteral.html)
  - We will be switching from random strings to reusable constant variables

**Description:**
Using raw strings when using the firebase functions to call on collections and document fields can lead to failures due to accidental typos. Using constants to represent those strings stored in a Constants java file will keep us safe from these accidental typos.

**Relevant Files:**
- [Constants.java](https://github.com/Landry333/Big-Owl/blob/cf99b99255f3a0251fadb5d7147b100c339a9050/app/src/main/java/com/example/bigowlapp/utils/Constants.java)
  - This is where we store constants

### PR DESCRIPTION
**Pull request by refatK**
### Related Issues
Closes #82

### Description
When referring to database fields, they are now stored as constants. This way, we have one source of error and also refactoring any time we want to change a field name rather than having to look through the code for all places a field is mentioned.

I also made a little python script if we ever add any new database collections or document types. You simply copy/paste the section of the class where you declare the variables and it will give you all the constants:
https://repl.it/@refatk17/ClassAttributesToFieldConstants#main.py

### How to check PR
Since this is a refactor PR, there isn't really anything to check other than the fact that the app should be working as it always has. More attention should be put on the implementation of the constants and any issues in the changes I made.

### Important Files:
- changed classes in `app/src/main/java/com/example/bigowlapp/model/` path

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/model/Group.java [hunk 1]
```diff
@@ -55,4 +54,13 @@ public void setMemberIdList(List<String> memberIdList) {
...
 
+    public static class Field {
+        public static final String NAME = "name";
+        public static final String SUPERVISOR_ID = "supervisorId";
+        public static final String MEMBER_ID_LIST = "memberIdList";
+
+        private Field() {
+            // constants class should not be instantiated
+        }
+    }
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/model/Group.java:65]: Very good way to store constants.

I prefer this way than keeping the constants in an outer class.

File: app/src/main/java/com/example/bigowlapp/model/Schedule.java [hunk 2]
```diff
@@ -111,4 +111,20 @@ public void setLocation(GeoPoint location) {
...
+        public static final String MEMBER_LIST = "memberList";
+        public static final String START_TIME = "startTime";
+        public static final String END_TIME = "endTime";
+        public static final String LOCATION = "location";
+        public static final String USER_SCHEDULE_RESPONSE_MAP = "userScheduleResponseMap";
+
+        private Field() {
+            // constants class should not be instantiated
+        }
+    }
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/model/Schedule.java:136]: Again, that's pretty smart.

**Review by ShawnGuerra-Bautista — APPROVED** [review]
Excellent.

**Comment by darkheros12** [comment]
I have checked every files. It is a good change. Anything deal with the database should be refer to a constant. It would be easier to read and more cohesion for the database. I will approve this PR after updating from latest branch.

**Review by darkheros12 — APPROVED** [review]
As mentioned in the previous comment.

**Comment by refatK** [comment]
I'll need to reupdate this one given the PRs that were merged in between, so it's in draft mode for now.

**Review by darkheros12 — APPROVED** [review]
As previously stated, the update has been made.

**Review by ShawnGuerra-Bautista — APPROVED** [review]
Just like before, everything works fine.
Great job.

**Pull request merged**